Volume/Views/MainView and Tabs/BookmarksView.swift

```diff
@@ -97,31 +105,83 @@ struct BookmarksView: View {
     // MARK: - Sections
 
     private var flyerContent: some View {
-        Group {
-            if viewModel.hasSavedFlyers {
-                switch viewModel.flyers {
-                case .none:
-                    ForEach(0..<4) { _ in
-                        FlyerCellPast.Skeleton()
-                            .padding(.bottom, 16)
-                    }
-                case .some(let flyers):
-                    ForEach(flyers) { flyer in
-                        if let urlString = flyer.imageUrl?.absoluteString {
-                            FlyerCellPast(
-                                flyer: flyer,
-                                navigationSource: .bookmarkFlyers,
-                                urlImageModel: URLImageModel(urlString: urlString),
-                                viewModel: FlyersView.ViewModel()
-                            )
+        VStack {
+            flyerSection(.upcoming)
+            flyerSection(.past)
+        }
+    }
+
+    private func flyerSection(_ flyerSection: FlyerSection) -> some View {
+        Section {
+            let selectedFlyers = flyerSection == .upcoming ? viewModel.upcomingFlyers : viewModel.pastFlyers
+
+            if let selectedFlyers, selectedFlyers.isEmpty {
+                VolumeMessage(
+                    image: Image.volume.flyer,
+                    message: .noBookmarkedFlyers,
+                    largeFont: true,
+                    fullWidth: true
+                )
+                .padding(EdgeInsets(top: 32, leading: 0, bottom: 48, trailing: 0))
+            } else {
+                ScrollView(.horizontal, showsIndicators: false) {
```

Comment:
Instead of having 2 statically sized sections, could we instead just do 1 vertical stack of the "Upcoming" label (if there are upcoming flyers), any upcoming flyers, "Past label" (if there are past flyers), and any past flyers? Then if both sections are empty, display the same empty state message as before.

Reply:
Ask design. I was just following the Figma.

Reply:
Approved by design. Please make changes when you're free